DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/README.md

```diff
@@ -0,0 +1,344 @@
+
+The source code for the the Eigenvalues and Eigenvectors Computation kernel can be found in `streaming_eigen.hpp` in the shared [include](../../include) directory. An detailed explanation of the algorithm of computing eigen values can be found [here](../pca/README.md#eigen-values-and-eigen-vectors-computation) in the PCA sample.
+
+### Construct output from Eigenvalues and Eigenvectors
+The final outputs of SVD can be constructed from the eigenvalues and eigenvectors of the input covariance. 
+
+Consider an input matrix A of size m x n.
+
+#### V Matrix
+The right singular vectors (the V matrix) can be produced by simply copying the eigenvectors , and should be of size n x n:
+```math
+V = [v_0, v_1, v_2, ...,v_n] 
+```
+
+#### S Matrix
+The singular value matrix (the S matrix) can be constructed as an m x n diagonal matrix (all but the values on the main diagonal are zeros) where the non-zero values are the square root of each eigenvalue:
+```math
+S =
+\begin{bmatrix}
+\sqrt{\lambda_{0}} & 0                  & 0                  & ... & 0 \\
+0                  & \sqrt{\lambda_{1}} & 0                  & ... & 0 \\
+0                  & 0                  & \sqrt{\lambda_{2}} & ... & 0 \\
+...                & ...                & ...                & ... & 0 \\
+0                  & 0                  & 0                  & 0   & \sqrt{\lambda_{n}} \\
+0                  & 0                  & 0                  & 0   & 0
+\end{bmatrix} 
+```
+
+#### U Matrix
+The left singular vectors (the U matrix) cannot be directly constructed with eigenvalues and eigenvectors of the input covariance. However, since we have now computed the U and S matrix, and knowing the original input A matrix, we can calculate U matrix using the original SVD relationship:
+```math
+A = USV^T
+```
+Since the V matrix is orthogonal by construction, $V^{-1}$ is the same as $V^T$, therefore:
+```math
+AV = US
+```
+And since S is a diagonal matrix, $S^{-1}$ is the same as multiplying by the reciprocals of each element on the main diagonal of $S$.
+
+So to sum up, the m x n portion of Matrix U (U[0:m][0:n]) can be obtained through:
+![](assets/U_matrix.png)
+
+When the input matrix A is not square, the number of eigenvectors calculated is less than the number of columns in the U matrix. An orthogonalization kernel is needed to generate extra orthogonal vectors.
+
+### U Matrix orthogonalization
+As mentioned above, when extra filler vectors are needed to complete the U matrix, we need to orthogonalize the matrix.
+
+An efficient algorithm to do this is already implemented in our [QR Decomposition sample](../qrd/README.md). So here we will just insert an instance of the `fpga_linalg::streamingQRD` design. The Q output of this kernel is orthogonalized U matrix. And the Q output is not used in the SVD design.
```

Comment:
```suggestion
An efficient algorithm to do this is already implemented in our [QR Decomposition sample](../qrd/README.md), so here we will insert an instance of the `fpga_linalg::streamingQRD` design. The $Q$ output of this kernel is orthogonalized $U$ matrix. Since we only care about the orthogonalized $U$ matrix, $R$ output of the Streaming QRD kernel is discarded.
```